test/groovy/com/sap/piper/integration/TransportManagementServiceTest.groovy

```diff
@@ -62,12 +62,69 @@ class TransportManagementServiceTest extends BasePiperTest {
         def clientSecret = 'mySecret'
 
         def tms = new TransportManagementService(nullScript, [verbose: true])
-        tms.authentication(uaaUrl, clientId, clientSecret)
+        def token = tms.authentication(uaaUrl, clientId, clientSecret)
 
         assertThat(loggingRule.log, containsString("[TransportManagementService] OAuth Token retrieval started."))
         assertThat(loggingRule.log, containsString("[TransportManagementService] UAA-URL: '${uaaUrl}', ClientId: '${clientId}'"))
         assertThat(loggingRule.log, containsString("Received response '{ \"access_token\": \"myOAuthToken\" }' with status 200."))
         assertThat(loggingRule.log, containsString("[TransportManagementService] OAuth Token retrieved successfully."))
+        assertThat(token, is('myOAuthToken'))
+    }
+
+    @Test
+    void retrieveOAuthToken__failure() {
+        Map requestParams
+        helper.registerAllowedMethod('httpRequest', [Map.class], { m ->
+            requestParams = m
+            return [content: 'Here an error message is expected', status: 400]
+        })
+
+        def uaaUrl = 'http://dummy.com/oauth'
+        def clientId = 'myId'
+        def clientSecret = 'mySecret'
+
+        def tms = new TransportManagementService(nullScript, [:])
+        def errorCaught = false
+        try {
+            tms.authentication(uaaUrl, clientId, clientSecret)
+        } catch (e) {
+            errorCaught = true
+            assertThat(e.getMessage(), is("OAuth Token retrieval failed. Consider re-running in verbose mode in order to get more details."))
+        } finally {
+            assertThat(errorCaught, is(true))
+            assertThat(loggingRule.log, containsString("[TransportManagementService] OAuth Token retrieval started."))
```

Comment:
Logging rule is prepared for working in the same way like the `ExpectedException` rule:

`loggingRule.expect(message)`. See e.g. [here](https://github.com/SAP/jenkins-library/blob/master/test/groovy/TransportRequestUploadFileTest.groovy#L159).

With that it is possible to  drop that way of catching the exception, do some checks in finally ...